### PR TITLE
LEAF-3435: Combined Inbox Implementation

### PR DESCRIPTION
**Pull request by raphaelfretter**

### CODE REVIEW

**Review by shaneodd — APPROVED** [review]

**Review by jrodcode — APPROVED** [review]

**Review by jampaul3 — APPROVED** [review]

**Review by aerinkayne — APPROVED** [review]

File: LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl [hunk 1]
```diff
@@ -1,337 +1,609 @@
...
+        // get data for any custom fields
+        let getData = [];
+        if (site.columns != undefined) {
+            for (let i in site.columns) {
+                let cols = site.columns[i].split(',');
+                for (let j in cols) {
+                    if (!isNaN(parseInt(cols[j]))) {
+                        getData.push(parseInt(cols[j]));
+                    } else {
+                        switch (cols[j]) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl:419]: stringtolower()

File: LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl [hunk 2]
```diff
@@ -1,337 +1,609 @@
...
-                                to[nextKey] = nextSource[nextKey];
-                            }
-                        }
-                    }
+    const getMapSites = new Promise((resolve, reject) => {
+        $.ajax({
+            type: 'GET',
+            url: './api/site/settings/sitemap_json',
+            success: function(res) {
+                if (res === 'Admin access required') {
```
Comment (Pelentan) [inline LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl:499]: This comparison worries me.  Not only does it have capitols, but spaces as well.

**Review by Pelentan — CHANGES_REQUESTED** [review]

File: LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl (hunk 2 above)
Comment (raphaelfretter) [reply LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl:499]: In the php file, if the user isn't in the admin group, it returns this specific message. I can alter it to return an int instead if that would be better.

**Review by maxrichardjr — APPROVED** [review]

**Review by AmandaSumner — APPROVED** [review]

**Review by pete-nerantzinis — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — COMMENTED** [review]
Looks good

**Pull request merged**